src/libraries/System.Text.Json/gen/Reflection/TypeExtensions.cs

```diff
@@ -1,41 +1,83 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
 
 namespace System.Text.Json.Reflection
 {
     internal static class TypeExtensions
     {
-        public static string GetUniqueCompilableTypeName(this Type type) => GetCompilableTypeName(type, type.FullName);
+        public static string GetCompilableName(this Type type)
+        {
+            if (type.IsArray)
+            {
+                return GetCompilableName(type.GetElementType()) + "[]";
+            }
 
-        public static string GetCompilableTypeName(this Type type) => GetCompilableTypeName(type, type.Name);
+            string compilableName;
 
-        private static string GetCompilableTypeName(Type type, string name)
-        {
             if (!type.IsGenericType)
             {
-                return name.Replace('+', '.');
+                compilableName = type.FullName;
             }
+            else
+            {
+                StringBuilder sb = new();
```

Comment:
nit: could use StringBuilder ctor that takes `type.FullName` (here and other similar)

Reply:
I'll make a dedicated PR addressing this.